src/pages/Error/ErrorPage.tsx

```diff
@@ -0,0 +1,48 @@
+import Button from "@/components/_common/Button/Button";
+import Title from "@/components/_common/Title/Title";
+import styled from "styled-components";
+import { FlexCenter } from "@/styles/common";
+import { PAGE_URL } from "@/constants/url";
+import { useNavigate } from "react-router-dom";
+
+const ErrorFallback = ({ error }: { error: Error }) => {
+  const navigate = useNavigate();
+
+  const handleReload = () => {
```

Comment:
에러 페이지를 만들어, 각 에러가 생성될때마다 HOME 버튼을 클릭해 이전 페이지로 돌아갈 수 있게 해주었습니다.
이 과정에서 문제 상황을 발견했는데요..!

`getCartItems` 에서 에러가 발생할 경우 해당 ErrorPage로 잘 이동하였지만, 
삭제를 하는 `removeCartItem` 나 수량 변경을 하는 `patchCartItemQuantity` 의 경우 api 요청 에러가 나도 해당 ErrorPage로 이동하지 않는 것을 확인했습니다😥

![스크린샷 2024-05-18 오후 5 24 19](https://github.com/woowacourse/react-shopping-cart/assets/125418818/b38e2259-51fe-406b-b989-ed41b877781e)

분명 error 를 똑같이 throw 하고 있는데, 위의 경우에는 ErrorBoundary에서 에러를 못잡는 이유를 아실까요..?? 같은 컴포넌트내에서 ErrorBoundary를 해도 해당 에러는 잡히지 않는데 이유가 궁금합니다..
그리고 해당 에러는 list에서 발생한 에러가 아니기 때문에 ErrorPage로 이동시키는 것 보다는, toast 메세지로 에러가 일어났음을 알리는 것이 좋을까요??

Reply:
https://legacy.reactjs.org/docs/error-boundaries.html#introducing-error-boundaries

위 링크 섹션에서 아래 부분을 참고해서 에러 바운더리가 잡을 수 있는 에러의 종류, 어떤 방식으로 에러를 탐지하는지 확인해보시면 좋겠습니다~

```
Note

Error boundaries do not catch errors for:

Event handlers ([learn more](https://legacy.reactjs.org/docs/error-boundaries.html#how-about-event-handlers))
Asynchronous code (e.g. setTimeout or requestAnimationFrame callbacks)
Server side rendering
Errors thrown in the error boundary itself (rather than its children)
```